src/pip/_internal/req/req_uninstall.py

```diff
@@ -183,6 +183,93 @@ def compress_for_output_listing(paths):
     return will_remove, will_skip
 
 
+class StashedUninstallPathSet(object):
+    """A set of file rename operations to stash files while
+    tentatively uninstalling them."""
+    def __init__(self):
+        self._save_dirs = {}
```

Comment:
It would be good to add a code comment here documenting the structure of these two attributes.